aten/src/ATen/native/cuda/ForeachBinaryOpList.cu

```diff
@@ -248,4 +250,44 @@ FOREACH_BINARY_OP_LIST(
     power_functor,
     /*division_op*/ true);
 
+template <typename T>
+struct Identity {
+  __device__ __forceinline__ T operator()(const T& x) {
+    return x;
+  }
+};
+
+void foreach_tensor_copy_list_kernel_cuda_(
+    TensorList self,
+    TensorList src,
+    const bool non_blocking) {
+  check_foreach_api_restrictions(self, src);
+  if (!can_use_fast_route(
+          self, src, /* does_op_promote_integer_inputs_to_float */ false)) {
+    return at::native::foreach_tensor_copy_list_kernel_slow_(
+        self, src, non_blocking);
+  }
+
+  std::vector<std::vector<at::Tensor>> tensor_lists{src.vec(), self.vec()};
+
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(
+      ScalarType::Half,
+      ScalarType::BFloat16,
+      ScalarType::Bool,
+      self[0].scalar_type(),
+      "foreach_tensor_copy",
+      [&]() {
+        using opmath_t = at::opmath_type<scalar_t>;
```

Comment:
This is for my understanding: I am not sure I wrapped my head correctly here. From what it looks like, it looks like we put the src and the thing we're copying from into a list of lists of Tensors...and then we call Identity on all of them with...a UNARY op? Shouldn't this be a binary op?

Reply:
Binary op functors take two or three tensorlists and require the given operator to take two arguments. The signature of inplace foreach copy does look binary bit the operator it calls is unary as it takes one argument and forwards it.